plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcConfig.java

```diff
@@ -90,4 +96,29 @@ public BaseJdbcConfig setCacheMissing(boolean cacheMissing)
         this.cacheMissing = cacheMissing;
         return this;
     }
+
+    @Min(1)
+    public long getCacheMaximumSize()
+    {
+        return cacheMaximumSize;
+    }
+
+    @Config("metadata.cache-maximum-size")
+    @ConfigDescription("Maximum number of objects stored in the metadata cache")
+    public BaseJdbcConfig setCacheMaximumSize(long cacheMaximumSize)
+    {
+        this.cacheMaximumSize = cacheMaximumSize;
+        return this;
+    }
+
+    @PostConstruct
+    public void validate()
+    {
+        if (metadataCacheTtl.equals(CACHING_DISABLED) && cacheMaximumSize != BaseJdbcConfig.DEFAULT_METADATA_CACHE_SIZE) {
+            throw new IllegalArgumentException(
+                    format("Invalid JDBC metadata caching configuration. No TTL is configured (%s) so metadata caching is disabled and changing the cache size to %s has no effect",
```

Comment:
```suggestion
                    format("metadata.cache-ttl must be set to a non-zero value when metadata.cache-maximum-size is set",
```

Maybe this? With an exception the actual property names causing issue should be present IMO.

Reply:
Yes, I like your suggestion as it is more concise. I went through a few iterations of this error message. Its hard to come up with a good error message :)